src/parse.rs

```diff
@@ -135,10 +139,7 @@ where
     // place and instead we collect LexErrors in `self.lex_errors` and attempt to continue parsing.
     // Once parsing has returned we want to surface an error in the order:
     // 1) LexError, 2) ParseError, 3) More Tokens Left
-    fn ensure_no_errors(
-        &mut self,
-        statements: Result<Vec<UntypedStatement>, ParseError>,
-    ) -> Result<Vec<UntypedStatement>, ParseError> {
+    fn ensure_no_errors<A>(&mut self, statements: Result<A, ParseError>) -> Result<A, ParseError> {
```

Comment:
I think we may need to now inject the error message below? It talks about statements but it could be an expr that is expected

Reply:
Ah needed to change the var name from `statements` to `parse_result`